### PR TITLE
Add gRPC metadata service to the ZenML helm chart

### PR DESCRIPTION
**Pull request by stefannica**
## Describe changes
The gRPC metadata service is a service that intermediates the access to the MLMD database and can speed up the pipeline runs especially when the client and/or orchestrator are not located in the same area as the MLMD database and need to connect to it over a network with high latency, like the internet.

This PR also:
- fixes a bug that prevented the use of client certificates in the database connection
- improves the helm chart with automatic reusing of self-signed certificates and other configuration values used during installation

## Pre-requisites
Please ensure you have done the following:
- [ ] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/component-gallery/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

File: docker/zenml-server-devel.Dockerfile [hunk 1]
```diff
@@ -1,30 +0,0 @@
...
-COPY README.md pyproject.toml ./
-# The existence of this __init__.py file allows the pip install before actually 
-# copying our source files which would invalidate caching
-COPY src/zenml/__init__.py ./src/zenml/
-
-RUN pip install -e .[server]
-COPY src src
-
-RUN mkdir -p .zenconfig/local_stores/default_zen_store
-COPY zenml.db .zenconfig/local_stores/default_zen_store
```
Comment (safoinme) [inline docker/zenml-server-devel.Dockerfile:23]: Sorry, why this file is deleted?

**Review by safoinme — APPROVED** [review]
I'm not really familiar with changes in this PR! but it looks good to me

File: docker/zenml-server-devel.Dockerfile (hunk 1 above)
Comment (stefannica) [reply docker/zenml-server-devel.Dockerfile:23]: it's not needed anymore. It has been replaced by `zenml-server-dev.Dockerfile`

**Review by wjayesh — APPROVED** [review]
LGTM!

File: src/zenml/zen_server/deploy/terraform/recipes/aws/zen_server.tf [hunk 2]
```diff
@@ -13,7 +13,11 @@ resource "helm_release" "zen-server" {
 
 
   set {
-    name = "image.tag"
+    name = "zenml.image.tag"
+    value = var.zenmlserver_image_tag
+  }
+  set {
+    name = "zenml.initImage.tag"
     value = var.zenmlserver_image_tag
```
[the remark this answers is not in the document]
Comment (stefannica) [reply src/zenml/zen_server/deploy/terraform/recipes/aws/zen_server.tf:21]: @wjayesh sure, they can be different. I just reused the same image tag variable here for convenience.

**Pull request merged**